server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -6,86 +6,82 @@
+        return self.mappings[section][option]
 
 
 class TestProcessTb:
     def test_token(self, monkeypatch, test_logger):
-        def mock_get(self, section: str, option: str) -> str:
-            assert option in [
-                "pbench-receive-dir-prefix",
-                "put-token",
-            ], f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
-            return fixture_path if option == "pbench-receive-dir-prefix" else ""
+        def mock_is_dir(self) -> bool:
+            return True
 
         expected_error_msg = (
             "No value for config option put-token in section pbench-server"
         )
-        monkeypatch.setattr(MockConfig, "get", mock_get)
+        mappings = {
+            "pbench-server": {
+                "pbench-receive-dir-prefix": receive_dir,
+                "put-token": "",
+            }
+        }
+        monkeypatch.setattr(Path, "is_dir", mock_is_dir)
         with pytest.raises(ValueError) as e:
-            ProcessTb(MockConfig(), test_logger)
+            ProcessTb(MockConfig(mappings), test_logger)
 
         assert expected_error_msg in str(e)
 
-    def test_get_receive_dir_failed(self, monkeypatch, test_logger):
-        def mock_get(self, section: str, option: str) -> str:
-            assert (
-                option == "pbench-receive-dir-prefix"
-            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
-            return ""
-
+    def test_get_receive_dir_failed(self, test_logger):
         expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
-        monkeypatch.setattr(MockConfig, "get", mock_get)
+        mappings = {
+            "pbench-server": {
+                "pbench-receive-dir-prefix": "",
+                "put-token": "Authorization-token",
+            }
+        }
         with pytest.raises(ValueError) as e:
-            ProcessTb(MockConfig(), test_logger)
+            ProcessTb(MockConfig(mappings), test_logger)
 
         assert expected_error_msg in str(e)
 
-    def test_get_receive_dir_value_failed(self, monkeypatch, test_logger):
-        def mock_get(self, section: str, option: str) -> str:
-            assert (
-                option == "pbench-receive-dir-prefix"
-            ), f"Unexpected value in option, {section!r}/{option!r}, in mocked config.get()"
-            return fixture_path / "wrong-directory"
-
+    def test_get_receive_dir_value_failed(self, test_logger):
         error_msg = "does not exist, or is not a directory] "
         expected_error_msg = error_msg + str(
-            fixture_path.resolve() / "wrong-directory-002"
+            receive_dir.resolve() / "wrong-directory-002"
         )
-
-        monkeypatch.setattr(MockConfig, "get", mock_get)
+        mappings = {
+            "pbench-server": {
+                "pbench-receive-dir-prefix": receive_dir / "wrong-directory",
```

Comment:
We were discussing before about using string literals and DRY -- it looks like you've fixed nearly all of them, but I see repetition in lines 61 and 65.

It's not a huge deal, but it would be better if you composed the `"wrong-directory-002"` from a variable which contains `"wrong-directory"` (which you would use at line 65 as well).  (You're already doing something similar in the other tests.)

And, you should consider whether you need and want the `resolve()` invocation.  That is, since you control `receive_dir`, and since it's not actually in the file system, you can set it to a "pre-resolved" value, which means that you can put the whole expression `receive_dir / "wrong-directory"` into a single variable.  On the other hand, if you _want_ to test the CUT's resolution of the directory path, then you should set `receive_dir` to a value which _won't_ be the same as `receive_dir.resolve()` (and maybe add a comment pointing out what the test code is doing there), so that the CUT's transposition will be evident.

Reply:
Also, you've made an interesting choice here to _not_ mock `is_dir()` in this test.  That means that you're counting on that directory actually not existing in the execution environment.  While that's a pretty good bet, it would be better to mock it anyway, so that the test can control and ensure the proper result and behavior.

Also, there's no need here for the `"pbench-receive-dir-prefix"` to include `receive_dir`, is there?  So, you can simplify (or, at least, "decouple") this test by using prefix which is specific to this test.

Reply:
> It's not a huge deal, but it would be better if you composed the "wrong-directory-002" from a variable which contains "wrong-directory" (which you would use at line 65 as well). (You're already doing something similar in the other tests.)

I made this change in a new commit.

> you can put the whole expression receive_dir / "wrong-directory" into a single variable

I adressed this

> On the other hand, if you want to test the CUT's resolution of the directory path, then you should set receive_dir to a value which won't be the same as receive_dir.resolve()

Unable to understand this

Reply:
> > On the other hand, if you want to test the CUT's resolution of the directory path, then you should set receive_dir to a value which won't be the same as receive_dir.resolve()
> 
> Unable to understand this

My apologies!

In the tests, you have code like the following:
```
    def test_process_tb_file_not_found_error(self, monkeypatch, caplog, test_logger):
        """checks normal processing when tar ball is not present"""
        receive_path = Path(TestProcessTb.receive_dir + "-002").resolve()
[...]
        mappings = {
            "pbench-server": {
                "pbench-receive-dir-prefix": TestProcessTb.receive_dir,
                "put-token": "Authorization-token",
            }
        }
```
So, the test is providing the value directly from `receive_dir` to the CUT as the prefix, but it is expecting the path which is used by the CUT to be suffixed with `-002` and to be "resolved", so it replicates those steps, putting the result in `receive_path` for later use.  Adding the suffix is fine (and necessary); however, using `resolve()` is interesting.

For many file paths, particularly ones which don't actually exist, the result of `resolve()` is the same as its input.  So, you can simplify the test code by selecting an appropriate value for `receive_dir` such that the test doesn't need to perform the `resolve()` step (because it won't change anything).

Alternatively, you can select a value for `receive_dir` which you know **will** be different from the value returned by `resolve()`, and by doing so the test can confirm that the CUT is using `resolve()` appropriately (but, this will look slightly strange in the code, so it will be worth a comment).

And, it's not a "one size fits all" thing -- you have the option of using the latter approach on only one of the tests and using the simple approach on all the others (counting on the one test to catch any problems).

Reply:
Thank you for explaining further. I thought of using the first method from the two suggested. I hope I understood the point clearly and added a commit.

I added an absolute path so that we won't have to go through resolve to complete the relative path.